server/internal/webrtc/webrtc.go

```diff
@@ -57,6 +57,11 @@ func (manager *WebRTCManager) Start() {
 
 	go func() {
 		for {
+			if manager.capture.Audio().GetSampleChannel() == nil {
+				// Pipeline not yet initialized
+				time.Sleep(50 * time.Millisecond)
```

Comment:
This might be race condition waiting to happen.

Reply:
Why?
There should only be one instance of an WebRTCManager, right?

The pipeline with the channel only gets created after the first user.
The other solution would be to move the channel into the CaptureManager, so it gets initialized earlier.

Reply:
Yes, currently it poses no issue. I was just thinking in the future when we would have simulcast or similar.

Or maybe starting the goroutine when sample channel is available - so with pipeline. But this is actually consumer of the samples, so it would be maybe better to have channel as consumer within webrtc manager and caputre would push to that channel.

So instead of pipeline when starting providing channel it would accept one. I need to try out this solution if this would be doable.

Reply:
It would be doable. I just removed a channel from the manager, where the pipeline wrote into :)

Reply:
Yeah, sorry for the confusion. I'll try to wrap my head around this and come back to you.

Reply:
Maybe `GetSampleChannel` could wait until pipeline is ready (again).

Reply:
I created the pipeline channel in the manager and pass it through.

This should work without the wait timer.